### PR TITLE
modify DataServlet.java, script.js and pom.xml to implement Json

### PR DESCRIPTION
**Pull request by Andrewmob2703-zz**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: please remove unnecessary space or empty line:)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You can refer to the open-sourced Google Java style guide for more thorough reference, but in general, in Java, we would use camel case, e.g., "toJson".

The same also applies to method names, e.g., prefer convertToJson(). About the name part "Gson" in the method name, you could keep it like "convertToJsonWithGson" if you think using Gson is something particularly important for the caller to be aware of; otherwise, prefer a shorter name without exposing the underlying implementation details with "convertToJson".

In general, we push for a consistent style guide to make it easy for everyone to read the code more easily. This is especially important when you work with a big team. Also, code is read way more often than written.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (Andrewmob2703-zz) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: unnecessary space removed

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: usually if a local variable is not used elsewhere except in a return statement, it'd be good to avoid it in the first place so that readers don't need to remember one more variable. Just do:

return gson.toJson(arrayList);

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: (1) please use camel case with a starting lower case for variable; (2) please use a more specific name for variables, e.g., wordArray, or something else if you prefer.

Also ArrayList is a core Java class. Even for a class name, you would want to avoid defining a class with this exact name.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Same here for method argument. Use another name and camel case.

File: portfolio/src/main/webapp/script.js
Comment (peizhaoou) [inline portfolio/src/main/webapp/script.js]: For now, it may be fine to use an array with hard-coded size, especially when you just started.

Going forward, try to think how you could avoid this. Think about the situation where the server side may be updated to output more objects, then your client code needs to change. Not to mention if object size is dynamic.

File: portfolio/src/main/webapp/script.js
Comment (peizhaoou) [inline portfolio/src/main/webapp/script.js]: Nit: may be good to use a slightly more concrete name for variables, even if this is Javascript. Consider something like "wordsArray", etc.

**Review by peizhaoou — COMMENTED** [review]
Thanks Andrew! Great job for making the client & server side working.

Don't be scared by my comments:) Many of them are tips about formats and getting better naming, which is something you would get better. Keep it up, it'll be just a matter of time to get these habits.

**Comment by Andrewmob2703-zz** [comment]
> Thanks Andrew! Great job for making the client & server side working.
> 
> Don't be scared by my comments:) Many of them are tips about formats and getting better naming, which is something you would get better. Keep it up, it'll be just a matter of time to get these habits.

No I appreciate these feedback. Please keep them coming...

**Review by peizhaoou — APPROVED** [review]
Thanks Andrew for making the change. Now that it all looks more of nitpicks that you hopefully already know how to address, I will approve it.

(Note that this may be my own personal preference. I generally approve a change if it overall looks good and only small (obvious) comments are left to address. My intention is that the code author then doesn't need to wait for another review round-trip time to get the approval. So I'd suggest fixing open comments before you submit even if you got approval. Have fun hacking!)

File: portfolio/src/main/webapp/script.js
Comment (martiKallas) [inline portfolio/src/main/webapp/script.js]: nit: This could probably have a better name that describes what it does. This becomes important when other people have to read and make changes to your code.

I'm not great at naming but my eaxmple would be, setGreeting or something similar. That way a future dev doesn't necessarily have to read the whole function to know what part of the code it is dealing with.

**Pull request merged**
